pinot-common/src/main/java/org/apache/pinot/common/metadata/ZKMetadataProvider.java

```diff
@@ -141,6 +141,26 @@ public static void removeResourceConfigFromPropertyStore(ZkHelixPropertyStore<ZN
     }
   }
 
+  /**
+   * Creates a new znode for SegmentZkMetadata. This call is atomic. If there are concurrent calls trying to create the
+   * same znode, only one of them would succeed.
+   *
+   * @param propertyStore Helix property store
+   * @param tableNameWithType Table name with type
+   * @param segmentZKMetadata Segment Zk metadata
+   * @return boolean indicating success/failure
+   */
+  public static boolean createSegmentZkMetadata(ZkHelixPropertyStore<ZNRecord> propertyStore, String tableNameWithType,
+      SegmentZKMetadata segmentZKMetadata) {
+    try {
+      return propertyStore
+          .create(constructPropertyStorePathForSegment(tableNameWithType, segmentZKMetadata.getSegmentName()),
+              segmentZKMetadata.toZNRecord(), AccessOption.PERSISTENT);
+    } catch (Exception e) {
+      return false;
```

Comment:
Could we add a message here when any exception is thrown from the ZK?

Reply:
I'm going to send up a followup PR for another change. I'll address this change there.